.github/workflows/node.yml

```
@@ -38,37 +55,85 @@ env:
+        # Which means that if the trigger was a label we run this job just if the label was Core changes
+        needs: check_should_run
+        if: needs.check_should_run.outputs.should_run == 'true'
+        outputs:
+          matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Load the engine matrix
+              id: load-engine-matrix
+              shell: bash
+              # We want to run the full matrix only for scheduled runs and Core changes
+              # For other events we want to run only the always matrix which is first and last version supported
+              run: |
+                if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "push" ]]; then
+                  echo "matrix=$(jq -c '[.[] | select(.run == "always")]' < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+                else
+                  echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+                fi
+    create-version-matrix:
+        runs-on: ubuntu-latest
+        needs: check_should_run
+        if: needs.check_should_run.outputs.should_run == 'true'
+        outputs:
+            matrix: ${{ steps.create-version-matrix.outputs.matrix }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Create version matrix
+              id: create-version-matrix
+              shell: bash
+              # We want to run the full matrix only for scheduled runs and Core changes
+              # For other events we want to run only the always matrix which is first and last version supported
+              run: |
+                if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "push" ]]; then
+                  echo 'matrix={"node":["16.x","20.x"]}' >> $GITHUB_OUTPUT
+                else
+                  echo 'matrix={"node":["16.x","17.x","18.x","19.x","20.x"]}' >> $GITHUB_OUTPUT
+                fi
 
-          - name: Load the engine matrix
-            id: load-engine-matrix
-            shell: bash
-            run: echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
 
     test-ubuntu-latest:
         runs-on: ubuntu-latest
-        needs: load-engine-matrix
+        needs: [load-engine-matrix, create-version-matrix]
         timeout-minutes: 25
         strategy:
             fail-fast: false
             matrix:
                 engine: ${{ fromJson(needs.load-engine-matrix.outputs.matrix) }}
-
+                node: ${{ fromJson(needs.create-version-matrix.outputs.matrix).node }}                
         steps:
             - uses: actions/checkout@v4
               with:
                   submodules: recursive
 
-            - name: Use Node.js 16.x
+            - name: Setup Node
               uses: actions/setup-node@v3
```

```suggestion
              uses: actions/setup-node@v4
```
This will remove warnings from report